packages/xforms-engine/src/lib/names/NamespaceDeclarationMap.ts

```diff
@@ -11,7 +11,7 @@ type NamedNodeDefinitionMap = ReadonlyMap<QualifiedName, NamedNodeDefinition>;
 export interface NamedSubtreeDefinition extends NamedNodeDefinition {
 	readonly namespaceDeclarations: NamespaceDeclarationMap;
 	readonly parent: NamedSubtreeDefinition | null;
-	readonly attributes?: NamedNodeDefinitionMap;
+	readonly attributes: NamedNodeDefinitionMap | null;
```

Comment:
Not a suggestion, just a comment: 
This is because LeafNode won't have attributes. When I see this, it makes me a bit hesitant to trust `attributes`, but I suppose it's okay since adding other subtrees like Group, Repeat, or LeafNode isn’t very common.

Reply:
Ah, I missed that `?` 🥲